src/index.ts

```diff
@@ -29,50 +31,75 @@ class CPK {
   static Call = OperationType.Call;
   static DelegateCall = OperationType.DelegateCall;
 
-  ethLibAdapter: EthLibAdapter;
-  transactionManager: TransactionManager;
+  appsSdk: SdkInstance;
+  safeAppInfo?: SafeInfo;
+
+  ethLibAdapter?: EthLibAdapter;
+  transactionManager?: TransactionManager;
   networks: NetworksConfig;
   ownerAccount?: Address;
+  isConnectedToSafe = false;
 
   contract?: Contract;
   multiSend?: Contract;
   proxyFactory?: Contract;
   masterCopyAddress?: Address;
   fallbackHandlerAddress?: Address;
 
-  isConnectedToSafe = false;
-  
-  static async create(opts: CPKConfig): Promise<CPK> {
-    if (!opts) throw new Error('missing options');
+  private setSafeInfo = (safeInfo: SafeInfo): void => {
```

Comment:
Would

```ts
private setSafeInfo(safeInfo: SafeInfo): void {
  // ...
}
```

still work here? It'd be consistent with the rest of the methods, and may be less confusing for `this` correctness.

Reply:
Reason why it is an arrow function is because it is used unbinded as a callback for the Safe SDK. Should investigate if `this` is the class or the instance...